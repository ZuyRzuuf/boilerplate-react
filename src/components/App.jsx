import React from 'react';

import Header from './Header';
import Dashboard from './Dashboard';

const App = () => {
    return (
        <>
            <Header />

            <Dashboard />
        </>
    );
};

export default App;
