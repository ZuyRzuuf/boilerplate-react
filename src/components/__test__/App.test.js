import React from 'react';
import { mount } from 'enzyme';

import App from '../App';

describe('A suite', function() {
    test('renders isolated component correctly', () => {
        const wrapper = shallow(
            <App />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders components tree correctly', () => {
        const wrapper = mount(
            <App />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should renders HTML tree correctly', () => {
        const html = render(
            <App />
        );
        expect(html).toMatchSnapshot();
    });
});
