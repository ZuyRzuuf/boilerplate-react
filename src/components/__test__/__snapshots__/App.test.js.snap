// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite renders components tree correctly 1`] = `
<App>
  <Header>
    <div>
      Backend/Frontend boilerplate without PHP
    </div>
  </Header>
  <Dashboard>
    <div>
      Dashboard
    </div>
  </Dashboard>
</App>
`;

exports[`A suite renders isolated component correctly 1`] = `
<Fragment>
  <Header />
  <Dashboard />
</Fragment>
`;

exports[`A suite should renders HTML tree correctly 1`] = `
Array [
  <div>
    Backend/Frontend boilerplate without PHP
  </div>,
  <div>
    Dashboard
  </div>,
]
`;
